api/models/ModerationAction.js

```diff
@@ -0,0 +1,146 @@
+    // email reportee and reporter
+    const group = await Group.find(groupId)
+    const reporter = await User.find(reporterId)
+    const post = await Post.find(postId)
+    const reportee = post.relations.user
+    const link = await Frontend.Route.post(this.get('post_id'), group)
+    const reporterLocale = reporter.get('settings')?.locale || 'en'
+    const reporteeLocale = reportee.get('settings')?.locale || 'en'
+
+    const messageReporter = type === 'created'
+      ? `${locales[reporterLocale].flaggedPostEmailContent({ post, group })}`
+      : `${locales[reporterLocale].clearedPostEmailContent({ post, group })}`
+
+    const messageReportee = type === 'created'
+      ? `${locales[reporteeLocale].flaggedPostEmailContent({ post, group })}`
+      : `${locales[reporteeLocale].clearedPostEmailContent({ post, group })}`
+
+    Queue.classMethod('Email', 'sendRawEmail', {
+      email: reporter.get('email'),
+      data: {
+        subject: locales[reporterLocale].moderatorClearedYouFlag(),
+        body: messageReporter +
+        `${link}\n\n`
+      }
+    })
+
+    Queue.classMethod('Email', 'sendRawEmail', {
+      email: reportee.get('email'),
+      data: {
+        subject: locales[reporteeLocale].moderatorClearedFlagFromPost(),
+        body: messageReportee +
+        `${link}\n\n`
+      }
+    })
+  },
+
+}, {
+  create: async function (data, opts) {
+    const { agreements, anonymous, platformAgreements, postId, groupId, reporterId, text } = data
+
+    const modAction = await ModerationAction.forge({ anonymous, post_id: postId, reporter_id: reporterId, text, status: 'active', group_id: groupId })
+      .save(null, pick(opts, 'transacting'))
+    await modAction.platformAgreements().attach(platformAgreements)
+    await modAction.agreements().attach(agreements)
+    await Post.addToFlaggedGroups({ postId, groupId })
+    return modAction
+  },
+
+  clearAction: async function ({ postId, groupId, moderationActionId }) {
+    let action
+    try {
+      action = await ModerationAction.where({ id: moderationActionId }).fetch()
+      await action.save({ status: 'cleared' }, { patch: true })
+    } catch (error) {
+      throw new Error('Moderation action not found')
+    }
+    await Post.removeFromFlaggedGroups({ postId, groupId })
+    return action
+  },
+
+  async sendToModerators ({ moderationActionId, postId, groupId, anonymous }) {
+    const moderationAction = await ModerationAction.find(moderationActionId)
+    const groups = await Post.find(postId).groups().fetch()
+
+    const send = async (group, userIds, anonymous) => {
+      const text = await moderationAction.getMessageText({ group, groupId, anonymous })
+      return sendMessageFromAxolotl(userIds, text)
+    }
+    for (let group of groups) {
+      const moderators = await group.moderators().fetch()
+      await send(group, moderators.map(moderator => moderator.id), anonymous)
+    }
+    const shouldSendToAdmins = (process.env.HYLO_ADMINS &&
```

Comment:
didn't we decide to only send to group moderators for now, not to us?

Reply:
So...

The current flagging setup actually sends us messages for certain report types. So I decided to replicate that here, since when we had the discussion it seems like we thought that we weren't actually getting messages from these flags